### DISCUSSION TITLE
strange issue with ScrollIntoView

### DISCUSSION THREAD

**Discussion by dornech** [GCuser99/SeleniumVBA#131] [General]
Lets assume I have a Webelement down the page I want to scroll to.
For testing/debugging I reference the VBA add-in noth the DLL.
In real code  I use findelement for this without object assignment but I omit this here to make it more clear because I think that should not not make the difference here, should it?

objWebElt.ScrollIntoView
-> does not work during regular execution, does only scroll by a few lines horizontally
-> does work in debug mode (stepping with F8)

objWebDriver.ExecuteScript "arguments[0].scrollIntoView({behavior: ""auto"", block: ""start"", inline: ""nearest""})", objWebElt
-> does work (regular execution & debugging/F8-stepping

objWebDriver.ExecuteScript "arguments[0].scrollIntoView()", objWebElt
-> does work (regular execution & debugging/F8-stepping

Strange, isn't it?

**Comment by dornech** [comment]
I did some research and some developers embbed the scrollintoview into a one-time timeout with setTimeout(callback, x) with x being in one example 0, in the other 50. Might be an idea in general, I already discovered that scrolling takes some time (execution seems not to be 100% syncronously i e executescript coming back after scroll is done and so this might help to avoid waits.
But unfortunately the current issue seems not to be resolved.

**Comment by GCuser99** [comment]
@dornech, I'm "driving blind" in trying to debug this issue without a reproducible use-case. Can you help me out? Otherwise it may take a while to construct one on my own. Will get back to you as soon as I can...

**Comment by dornech** [comment]
Unfortunately the problem occurs in a protected area with sensitive data, I tried my very best to explain it as generally as possible. 

I can overcome the problem in my case with a movetoelement action chain. This might be an option for others as well.

**Comment by GCuser99** [comment]
@dornech, does the following work on your system?
```vba
objWebElt.ScrollIntoView enSpeed:=jump_instant
```

**Comment by GCuser99** [comment]
I've concluded that this thread belongs in Discussions - this is an intrinsic problem with jscript's scrollIntoView when using smooth scrolling...

**Comment by GCuser99** [comment]
Resolved in v5.9

**Comment by dornech** [comment]
Unfortunately it seems to be an neverending story. I still have trouble and scrollIntoView does not work reliably.
However, I took a different approach based on ActionChain - this works much more reliable.
Code is pretty straight forward I think.
This might help others as well.

```
'------------------------------------------------------------------------------------
' MoveToElement utilities
' alternative to scollIntoView - help to overcome problems
'------------------------------------------------------------------------------------

Sub moveToElement(ByRef objWebDriver As SeleniumVBA.WebDriver, ByRef objWebElement As SeleniumVBA.WebElement, Optional b_Click As Boolean = False)

   Dim _
      objWebActionChain As SeleniumVBA.WebActionChain

   Set objWebActionChain = objWebDriver.ActionChain

   If Not b_Click Then
      objWebActionChain.moveToElement(objWebElement).Perform
      objWebDriver.Wait 100
   Else
      objWebActionChain.moveToElement(objWebElement).Click.Perform
      objWebDriver.Wait 250
   End If

   Set objWebActionChain = Nothing

End Sub


Sub moveToElementAndClick(ByRef objWebDriver As SeleniumVBA.WebDriver, ByRef objWebElement As SeleniumVBA.WebElement)

   Call moveToElement(objWebDriver, objWebElement, True)

End Sub



```

**Comment by GCuser99** [comment]
Sorry you are having problems with this again. I wish I knew more about the reliability issue. I have tested on a number of use cases. 

In the test_Scroll module, there are several test subs - do those seem to work for you?

Are the reliability issues only happening when there is a scroll animation involved? Did you try setting the scroll speed to `jump_instant` to avoid animations?

```vba
driver.ScrollIntoView element, enSpeed:=jump_instant 
element.ScrollIntoView enSpeed:=jump_instant  
```
Thanks for your patience but it is hard for me to debug when I don't have a failing use-case.

**Comment by dornech** [comment]
Hi, thanks for caring.
I just used ScrollIntoView with default parameters. To be honest - I didn't try different parameters because I developed the other approach earlier.
Funny part of the story (and this is what I mean with reliable): I use a macro to automate the download of PDF documents from my bank (which is also the reason I cannot provide the sample site). One account ScrollIntoView worked, another account it does not work. Same bank, same HTML code ... so I have not any kind of clue what's going on there.

**Comment by dornech** [comment]
I just wanted to share my experience and provide a workaround in case others have similar issues.

**Comment by GCuser99** [comment]
Ok. One more question - are you using Firefox or Chrome/Edge when ScollIntoView does not work?

**Comment by dornech** [comment]
Chrome. Years ago I was primarily working with Firefox but support of the earlier Selenium COM-Wrapper SeleniumBasic forced me to work with Chrome.
